producers/mysql-producer/src/main/java/com/flipkart/aesop/runtime/producer/avro/MysqlAvroEventManager.java

```diff
@@ -137,19 +143,39 @@ public int createAndAppendEvent(DbChangeEntry changeEntry, DbusEventBufferAppend
 		{
 			final long timestampInNanos = eventHeader.getTimestamp() * 1000000L;
 			final boolean isReplicated = false;
-			for (Row row : rowList)
+			for (Pair pair : listOfPairs)
 			{
-				List<Column> columns = row.getColumns();
+				Row oldRow = (Row) pair.getBefore();
+				Row newRow = (Row) pair.getAfter();
 				// getting the appropriate bin log mapper for the logicalSource
 				BinLogEventMapper<T> binLogEventMapper =
-				        binLogEventMappers.get(lSourceId) == null ? new DefaultBinLogEventMapper<T>(new ORToAvroMapper())
-				                : binLogEventMappers.get(lSourceId);
+						binLogEventMappers.get(lSourceId) == null ? new DefaultBinLogEventMapper<T>(new ORToAvroMapper())
+								: binLogEventMappers.get(lSourceId);
+				GenericRecord newRecord = binLogEventMapper.mapBinLogEvent(eventHeader, newRow, dbusOpCode, schema);
+
+				if (this.isOldValueRequired)
+				{
+					String rowChangeFieldName = AvroSchemaHelper.getRowChangeField(schema);
+					if (rowChangeFieldName == null)
+					{
+						LOGGER.error("Schema Configuration Mismatch: isOldValueRequired flag is set but no field in schema found");
+						throw new InvalidAvroSchemaException("isOldValueRequired flag is set but no field in schema found");
+					}
 
-				GenericRecord genericRecord = binLogEventMapper.mapBinLogEvent(eventHeader, row, dbusOpCode, schema);
-				List<KeyPair> keyPairList = generateKeyPair(columns, schema);
+					Map<String, Object> changedOldValues = null;
+					if(oldRow != null)
+					{
+						GenericRecord oldRecord = binLogEventMapper.mapBinLogEvent(eventHeader, oldRow, dbusOpCode, schema);
+						changedOldValues = MysqlAvroEventManager.calculateChange(oldRecord, newRecord);
+					}
+					newRecord.put(rowChangeFieldName, changedOldValues);
+				}
+
+				List<KeyPair> keyPairList = generateKeyPair(newRow.getColumns(), schema);
+				LOGGER.info("Final new record" + newRecord.toString());
```

Comment:
LOGGER.debug
Also, see if the logger SPI provides Slf4j like support to avoid the actual string concatenation overhead if the logging level is debug.